frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideInBranchBelowAction.java

```diff
@@ -0,0 +1,199 @@
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.Presentation;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsNotifier;
+import git4idea.branch.GitNewBranchDialog;
+import git4idea.repo.GitRepository;
+import io.vavr.collection.List;
+import io.vavr.control.Try;
+import lombok.CustomLog;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.branchlayout.api.BranchLayoutEntry;
+import com.virtuslab.gitmachete.frontend.actions.common.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.SlideInDialog;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyGitMacheteRepository;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyProject;
+import com.virtuslab.logger.IEnhancedLambdaLogger;
+
+@CustomLog
+public abstract class BaseSlideInBranchBelowAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IExpectsKeyProject,
+      IExpectsKeyGitMacheteRepository,
+      IBranchNameProvider {
+
+  @Override
+  public IEnhancedLambdaLogger log() {
+    return LOG;
+  }
+
+  @Override
+  @UIEffect
+  public void update(AnActionEvent anActionEvent) {
+    super.update(anActionEvent);
+
+    Presentation presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    var branchName = getNameOfBranchUnderAction(anActionEvent);
+    var branch = branchName.flatMap(bn -> getGitMacheteBranchByName(anActionEvent, bn));
+
+    if (branchName.isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(
+          GitMacheteBundle.message("action.GitMachete.BaseSlideInBranchBelowAction.description.disabled.no-parent"));
+    } else if (branch.isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(
+          GitMacheteBundle.message("action.GitMachete.description.disabled.undefined.machete-branch", "Slide In"));
+    } else {
+      presentation.setDescription(
+          GitMacheteBundle.message("action.GitMachete.BaseSlideInBranchBelowAction.description", branchName.get()));
+
+      if (getCurrentBranchNameIfManaged(anActionEvent).equals(branchName)) {
+        presentation.setText(GitMacheteBundle.message("action.GitMachete.BaseSlideInBranchBelowAction.text.current-branch"));
+      }
+    }
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    var project = getProject(anActionEvent);
+    var selectedVcsRepository = getSelectedGitRepository(anActionEvent).getOrNull();
+    var gitMacheteParentBranch = getNameOfBranchUnderAction(anActionEvent)
+        .flatMap(bn -> getGitMacheteBranchByName(anActionEvent, bn)).getOrNull();
```

Comment:
Potentially this mapping can be omitted: I know, it's for checking that selected branch is present in repo but (just thinking loud) if this branch is provided by (IIRC) `GraphTable` so this branch must exist in repo (at least in moment of repo refreshing) since it was displayed in graph, so I think we cant trust the data provided by `GraphTable`.
WDYT?

Reply:
It took me a while to understand your point but... XD yeah you are right